sympy/ntheory/generate.py

```diff
@@ -608,36 +618,41 @@ def prevprime(n):
         nextprime : Return the ith prime greater than n
         primerange : Generates all primes in a given range
     """
-    from sympy.functions.elementary.integers import ceiling
+    @classmethod
+    def eval(cls, n):
+        from sympy.functions.elementary.integers import ceiling
 
-    # wrapping ceiling in as_int will raise an error if there was a problem
-    # determining whether the expression was exactly an integer or not
-    n = as_int(ceiling(n))
-    if n < 3:
-        raise ValueError("no preceding primes")
-    if n < 8:
-        return {3: 2, 4: 3, 5: 3, 6: 5, 7: 5}[n]
-    if n <= sieve._list[-1]:
-        l, u = sieve.search(n)
-        if l == u:
-            return sieve[l-1]
+        # wrapping ceiling in as_int will raise an error if there was a problem
+        # determining whether the expression was exactly an integer or not
+        if not isinstance(n, Integer):
+            return None
+
+        n = as_int(ceiling(n))
```

Comment:
```suggestion
        n = ceiling(n)
```
What if n is symbol?